### ISSUE TITLE
Automatically handle errors

### ISSUE THREAD

**Issue by lukehoban** [pulumi/pulumi-ai#39]
It would be helpful for `pulumi-ai` to be able to automatically handle any error messages raised during deployment, instead of requiring the user to request that such errors are handled.  In the most common cases, the AI will know how to fix the errors that come up.

**Comment by mworks-proj** [comment]
yes, while using pulumi to create a lambda function i got this error: { The runtime parameter of nodejs12.x is no longer supported for creating or updating AWS Lambda functions. We recommend you use the new runtime (nodejs18.x) while creating or updating functions.",
  Type: "User" } it then prompts me to try "Fix the  error" - did but same error output.
